orchestrator/careplanservice/service_test.go

```diff
@@ -46,6 +53,100 @@ func TestService_Proxy(t *testing.T) {
 	require.Equal(t, fhirServerURL.Host, capturedHost)
 }
 
+type OperationOutcomeWithResourceType struct {
+	fhir.OperationOutcome
+	ResourceType *string `bson:"resourceType" json:"resourceType"`
+}
+
+// Test invalid requests return OperationOutcome
+func TestService_Post_Task_Error(t *testing.T) {
+	var tests = []struct {
+		method             string
+		path               string
+		body               string
+		expectedStatusCode int
+		expectedMessage    string
+	}{
+		{
+			"POST",
+			"/cps/Task",
+			"",
+			http.StatusBadRequest,
+			"CarePlanService/CreateTask failed: invalid Task: unexpected end of JSON input",
+		},
+		{
+			"PUT",
+			"/cps/Task/no-such-task",
+			"",
+			http.StatusBadRequest,
+			"CarePlanService/UpdateTask failed: invalid Task: unexpected end of JSON input",
+		},
+		{
+			"PUT",
+			"/cps/Task/no-such-task",
+			taskJSON,
+			http.StatusBadRequest,
+			"CarePlanService/UpdateTask failed: failed to read Task: FHIR request failed (GET http://example.com/Task/no-such-task, status=500)",
+		},
+		{
+			"POST",
+			"/cps/CarePlan",
+			"",
+			http.StatusBadRequest,
+			"CarePlanService/CreateCarePlan failed: invalid fhir.CarePlan: unexpected end of JSON input",
+		},
+	}
+
+	// Setup: configure the service
+	tokenIntrospectionEndpoint := setupAuthorizationServer(t)
+	service, err := New(
+		Config{
+			FHIR: FHIRConfig{
+				BaseURL: "http://example.com",
+			},
+		},
+		nutsPublicURL,
+		orcaPublicURL,
+		tokenIntrospectionEndpoint,
+		"did:web:example.com",
+		nil)
+	require.NoError(t, err)
+	serverMux := http.NewServeMux()
+	service.RegisterHandlers(serverMux)
+	server := httptest.NewServer(serverMux)
+
+	// Setup: configure the client
+	httpClient := server.Client()
+	httpClient.Transport = auth.AuthenticatedTestRoundTripper(server.Client().Transport, "")
+
+	for _, tt := range tests {
+		// Make an invalid call (not providing JSON payload)
+		request, err := http.NewRequest(tt.method, server.URL+tt.path, strings.NewReader(tt.body))
+		require.NoError(t, err)
+		request.Header.Set("Content-Type", "application/fhir+json")
+
+		httpResponse, err := httpClient.Do(request)
+		require.NoError(t, err)
+
+		// Test response
+		require.Equal(t, tt.expectedStatusCode, httpResponse.StatusCode)
+
+		defer func(Body io.ReadCloser) {
+			err := Body.Close()
+			require.NoError(t, err)
```

Comment:
no need to check this in the test, I'd think. If it fails, it indicates the test misbehaving (probably due to another assertion failure).

Actually, I don't think I ever close byte streams in tests (since they're so short-lived and then automatically cleaned up). So you could leave it out altogether.